server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java

```diff
@@ -59,7 +59,7 @@ public void testDisruptionOnSnapshotInitialization() throws Exception {
             .put(DiscoverySettings.COMMIT_TIMEOUT_SETTING.getKey(), "30s") // wait till cluster state is committed
             .build();
         final String idxName = "test";
-        configureCluster(settings, 4, 2);
+        configureCluster(settings);
```

Comment:
This is the only test case in this suite, so we could pass the settings to each node by overriding `nodeSettings` ourselves, and drop the `configureCluster()` call. That then means that this suite can just inherit directly from `ESIntegTestCase` - it doesn't use any of the other functionality that `AbstractDisruptionTestCase` provides.

Reply:
@DaveCTurner actually, I'm not so sure here. We're also using the `DEFAULT_SETTINGS` on `AbstractDisruptionTestCase`, maybe https://github.com/elastic/elasticsearch/pull/34808/commits/cd1ed8711f141158c5997d3626d1940329f9c7d4 isn't so bad afterall?

Reply:
Otherwise we're forced to duplicate the logic in org.elasticsearch.discovery.AbstractDisruptionTestCase#nodeSettings

Reply:
I meant this:

```diff
diff --git a/server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java b/server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java
index 7d4fcff2f47..e825e2f53cd 100644
--- a/server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java
+++ b/server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java
@@ -32,6 +32,7 @@ import org.elasticsearch.snapshots.SnapshotInfo;
 import org.elasticsearch.snapshots.SnapshotMissingException;
 import org.elasticsearch.snapshots.SnapshotState;
 import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.discovery.TestZenDiscovery;
 import org.elasticsearch.test.disruption.NetworkDisruption;
 import org.elasticsearch.test.junit.annotations.TestLogging;

@@ -50,11 +51,13 @@ import static org.hamcrest.Matchers.instanceOf;
  */
 @TestLogging("org.elasticsearch.snapshot:TRACE")
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0, transportClientRatio = 0)
-public class SnapshotDisruptionIT extends AbstractDisruptionTestCase {
+public class SnapshotDisruptionIT extends ESIntegTestCase {

     @Override
     protected Settings nodeSettings(int nodeOrdinal) {
         return Settings.builder().put(super.nodeSettings(nodeOrdinal))
+            .put(AbstractDisruptionTestCase.DEFAULT_SETTINGS)
+            .put(TestZenDiscovery.USE_MOCK_PINGS.getKey(), false)
             .put(DiscoverySettings.COMMIT_TIMEOUT_SETTING.getKey(), "30s")
             .build();
     }
```

Basically I don't like the statefulness of `currentSettings` and would prefer to be explicit about the settings we're using when starting up each node. Looking at the code in more detail I see that we only call `configureCluster` within `ClusterDisruptionIT`, in two places. In one place we call it to add settings, and I think it'd be better to add those settings when starting the nodes instead. In the other place we pass in `Settings.EMPTY`; however that entire test uses nothing else from `AbstractDisruptionTestCase` so maybe it should be in an `ESIntegTestCase` fixture on its own instead.